app/middlewares/vendor_api_request_middleware.rb

```diff
@@ -0,0 +1,57 @@
+require 'redis'
+require './app/workers/vendor_api_request_worker'
+
+class VendorAPIRequestMiddleware
+  REQUEST_HEADER_KEYS = %w[
+    HTTP_VERSION
+    HTTP_HOST
+    HTTP_USER_AGENT
+    HTTP_ACCEPT
+    HTTP_ACCEPT_ENCODING
+    HTTP_AUTHORIZATION
+    HTTP_CONNECTION
+    HTTP_CACHE_CONTROL
+  ].freeze
+
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    started_at = Time.zone.now
+
+    @request = Rack::Request.new(env)
+    status, headers, response = @app.call(env)
+
+    begin
+      if vendor_api_path?
+        finished_at = Time.zone.now
+        response_time = (finished_at - started_at) * 1000
```

Comment:
Will this be accurate if there are middleware that wrap this middleware?

Reply:
No it's not going to account for time taken by any subsequent middleware. @duncanjbrown asked the question (in the spike PR) of whether we needed to record this as Skylight should have a more accurate metric. Happy to remove this and the `response_time` db field if we think it's better to leave timings out.

Reply:
I think it's better to leave out the timings if they could be misleading

Reply:
See https://github.com/DFE-Digital/apply-for-teacher-training/pull/3029 for db amends